core/services/relay/evm/chain_reader_test.go

```diff
@@ -0,0 +1,345 @@
+												"sender": "0x0",
+												"to": "0x0"
+											},
+											"returnValues": [
+												"sender",
+												"amount0In",
+												"amount1In",
+												"amount0Out",
+												"amount1Out",
+												"to"
+											],
+											"readType": 1
+										}`,
+		})
+
+	testCases = append(testCases,
+		testCase{
+			name:     "functionWithOneParamAndMultipleResponses",
+			abiInput: `{"constant":true,"inputs":[{"internalType":"address","name":"_user","type":"address"}],"name":"getUserAccountData","outputs":[{"internalType":"uint256","name":"totalLiquidityETH","type":"uint256"},{"internalType":"uint256","name":"totalCollateralETH","type":"uint256"},{"internalType":"uint256","name":"totalBorrowsETH","type":"uint256"},{"internalType":"uint256","name":"totalFeesETH","type":"uint256"},{"internalType":"uint256","name":"availableBorrowsETH","type":"uint256"},{"internalType":"uint256","name":"currentLiquidationThreshold","type":"uint256"},{"internalType":"uint256","name":"ltv","type":"uint256"},{"internalType":"uint256","name":"healthFactor","type":"uint256"}],"payable":false,"stateMutability":"view","type":"function"}`,
+			chainReadingDefinitions: `"getUserAccountData":{
+											"chainSpecificName": "getUserAccountData",
+											"params":{
+												"_user": "0x0"
+											},
+											"returnValues": [
+												"totalLiquidityETH",
+												"totalCollateralETH",
+												"totalBorrowsETH",
+												"totalFeesETH",
+												"availableBorrowsETH",
+												"currentLiquidationThreshold",
+												"healthFactor"
+											],
+											"readType": 0
+										}`,
+		})
+
+	testCases = append(testCases,
+		testCase{
+			name:     "functionWithNoParamsAndOneResponseWithNoName",
+			abiInput: `{"inputs":[],"name":"name","outputs":[{"internalType":"string","name":"","type":"string"}],"stateMutability":"view","type":"function"}`,
+			chainReadingDefinitions: `"name":{
+											"chainSpecificName": "name",
+											"returnValues": [
+												""
+											],
+											"readType": 0
+										}`,
+		})
+
+	testCases = append(testCases,
+		testCase{
+			name:     "functionWithMultipleParamsAndOneResult",
+			abiInput: `{"inputs":[{"internalType":"address","name":"_input","type":"address"},{"internalType":"address","name":"_output","type":"address"},{"internalType":"uint256","name":"_inputQuantity","type":"uint256"}],"name":"getSwapOutput","outputs":[{"internalType":"uint256","name":"swapOutput","type":"uint256"}],"stateMutability":"view","type":"function"}`,
+			chainReadingDefinitions: `"getSwapOutput":{
+											"chainSpecificName": "getSwapOutput",
+											"params":{
+												"_input":"0x0",
+												"_output":"0x0",
+												"_inputQuantity":"0x0"
+											},
+											"returnValues": [
+												"swapOutput"
+											],
+											"readType": 0
+										}`,
+		})
+
+	// TODO how to handle return values for tuples
```

Comment:
When TODO? Can we link a ticket?

Reply:
@ilija42 If I recall, but "tuples" you mean something like... nested structs right?  I think the only thing we're missing is ability to rename nested struct fields, which should be what Ryan's work addresses.  Either way, I think we can remove this comment right?

Reply:
Added a ticket to this. This will probably require a more verbose way for expressing return values, similar to the one proposed as alternative design in the CLIP. I will sync up with Ryan to see what was his idea for Codec transformers and then adress this